src/api/index.ts

```diff
@@ -342,6 +359,68 @@ export const withdraw = async (code: TokenCode, amount: Balance, account?: Accou
   return DutchX.withdraw(code, amount, account)
 }
 
+/**
+* getSellerOngoingAuctions
+* @param tokensJSON
+* @param account
+*/
+export const getSellerOngoingAuctions = async (tokensJSON: DefaultTokenList, account: Account): Promise<any[]> => {
+  const { DutchX } = await promisedAPI
```

Comment:
```
export const getSellerOngoingAuctions = async (tokensJSON: DefaultTokenList, account: Account): Promise<any[]> => {
  const { DutchX } = await promisedAPI
  // assuming tokensJSON comes in form:
  // defaultToken = { name: 'Ether Token', address: '0xAg9823nfejcdksak1o38fFa09384', imgBytes: [ ... ] }
  const tokensJSONAddresses: Account[] = tokensJSON.map(t => t.address)
  const runningPairsArr = await DutchX.getRunningTokenPairs(tokensJSONAddresses)

  // get Array back of Auctions user is in
  // grab sellerBalance of USER for each current ongoing auction
  // @ts-ignore
  const sellerOngoingAuctions: number[] = (await DutchX.getSellerBalancesOfCurrentAuctions(...runningPairsArr, account)).map((res: any) => res.toNumber())
  // we know user is participating in (runningPairsArr[0][0]-runningPairsArr[1][0] && runningPairsArr[0][3]-runningPairsArr[1][3])

  // TODO: addressesToTokenJSON can be calculated once when we get the list from IPFS
  // or at least memoize calculation with reselect
  const addressesToTokenJSON = tokensJSON.reduce((acc, tk) => {
    acc[tk.name] = tk
    return acc
  }, {}) as {[P in TokenCode]: DefaultTokenObject}

  const [runningPairsS, runningPairsB] = runningPairsArr

  const promisedClaimableTokens: Promise<[BigNumber[], BigNumber[]]>[] = []
  const ongoingAuctions: {sell: DefaultTokenObject, buy: DefaultTokenObject}[] = sellerOngoingAuctions.reduce((accum, bal, index) => {
    if (!bal) return accum

    const s = runningPairsS[index]
    const b = runningPairsB[index]

    const sell = addressesToTokenJSON[s]
    const buy = addressesToTokenJSON[b]
    if (sell && buy) {
      accum.push({ sell, buy })
      promisedClaimableTokens.push(
        DutchX.getIndicesWithClaimableTokensForSellers(sell.address, buy.address, account, 0)
      )
    }

    return accum
  }, [])

  // Checks ongoingAuctions Array if each ongoingAuction has claimable Tokens
  // Array indices are lined up
  // @returns => forEach ongoingAuction => (indices[], userBalanceInSpecificAuction[]) => e.g for: ETH/GNO => (indices[], userBalance[])
  const claimableTokens = await Promise.all(promisedClaimableTokens)
  /* console.log(claimableTokens)
  console.log(`
    claimableTokens = ${claimableTokens}
    specific arr    = ${claimableTokens[0][0]}
    length          = ${claimableTokens[0][0].length}
  `) */

  // consider adding LAST userBalance from claimableTokens to ongoingAuctions object as COMMITTED prop
  const auctionsArray = ongoingAuctions.map((auction, index) => {
    const [indices, balancePerIndex] = claimableTokens[index]
    return {
      ...auction,
      indices,
      balancePerIndex,
      claim: indices.length >= 2,
    }
  })

  return auctionsArray
}
```
one less loop when gathering promises for `claimableTokens` and fewer iterations on lookup inside `tokensJSON`